pandas/core/arrays/arrow/array.py

```diff
@@ -553,6 +553,13 @@ def __getitem__(self, item: PositionalIndexer):
             )
         # We are not an array indexer, so maybe e.g. a slice or integer
         # indexer. We dispatch to pyarrow.
+        if isinstance(item, slice):
+            if item.start == item.stop:
+                pass
+            elif item.start == -len(self) - 1:
```

Comment:
should this be `item.start <= -len(self)`?

Not sure if you want to mimic numpy behavior here but numpy allows slicing beyond the len of the array:

```
In [1]: import numpy as np

In [2]: np.array(["a", "b", "c"])[slice(-10, 10)]
Out[2]: array(['a', 'b', 'c'], dtype='<U1')
```

Reply:
Yeah that makes sense, updated